function/expression.go

```diff
@@ -35,7 +35,7 @@ import (
 // from the backend systems.
 // * Contains current timerange being queried for - this can be
 // changed by say, application of time shift function.
-type EvaluationContext struct {
+type EvaluationContextBuilder struct {
```

Comment:
Fixed; also added some more comments and replaced `NewEvaluationContext(builder)` with `builder.Build()` for ergonomics.